### DISCUSSION TITLE
[Idea] Mikrotik SwOS SNMP support

### DISCUSSION THREAD

**Discussion by Mudzay** [glpi-project/glpi-agent#591] [Ideas]
### Your idea

GLPI-agent currently doesn't support SwOS devices as their OID is iso.1.3.6.1.4.1.14988.2
The [file](https://github.com/glpi-project/glpi-agent/blob/develop/lib/GLPI/Agent/SNMP/MibSupport/Mikrotik.pm) defining mikrotik devices defines mikrotik OID as iso.1.3.6.1.4.1.14988.1 which is correct only for devices running RouterOS.

**Comment by g-bougard** [comment]
Hi @Mudzay 

can you provide a [snmpwalk output in the right format](https://glpi-agent.readthedocs.io/en/latest/bug-reporting.html#snmpwalk-output) for such device ?

**Reply by Mudzay** [reply]
Sure, there's the file
[crs328-24p-4s+.txt](https://github.com/glpi-project/glpi-agent/files/14083805/crs328-24p-4s%2B.txt)

**Reply by g-bougard** [reply]
Hi @Mudzay 

thank you for the walk.

Anyway, I don't see what could be taken as serialnumber in the walk. Do you know it ? Have you the mib for that device ?

**Reply by Mudzay** [reply]
Now i see that serialnumber should be at 1.3.6.1.4.1.14988.1.1.7.3, but it's not present in this walk (not present in other SwOS switches, RouterOS switches provide their SN at this MIB)
Seems like there's no complete MIKROTIK-MIB support for SwOS :(
Anyway, there's [mikrotik-mib documentation](https://mibbrowser.online/mibdb_search.php?mib=MIKROTIK-MIB), maybe it can help in some way

**Reply by g-bougard** [reply]
Yes, this is still the OID the agent is searching for.

About the mib, I still found one few times ago when I wrote the [Mikrotik.pm module](https://github.com/glpi-project/glpi-agent/blob/develop/lib/GLPI/Agent/SNMP/MibSupport/Mikrotik.pm) and mine is still more recent: it has `202010080000Z` as `LAST-UPDATED` property and yours `201703100000Z`. Can you contact the Mikrotic support and ask for a more recent mib or better ask them how to get your device serialnumber via snmp ? ;-)

Without serialnumer, you may try to enable import and update rules by MAC.

**Comment by radityo** [comment]
Hi, I upgrade my CRS312-4C+8XG to 2.16 and in the change log there's this entry:

```*) added SNMP serial number and version OIDs;```

I check with snmpwalk and there are new entries: `.1.3.6.1.4.1.14988.1.1.7.3.0` and `.1.3.6.1.4.1.14988.1.1.7.4.0` . I attached it here:

[CRS312-4C+8XG.txt](https://github.com/glpi-project/glpi-agent/files/14442183/CRS312-4C%2B8XG.txt)

edit: forgot to mentioned that if I let glpi-agent scan its IP address, it will add some unmanaged device.

**Reply by g-bougard** [reply]
Hi @radityo 

thank you for the sharing, I'll try to update this device support.
